packages/appsync-modelgen-plugin/src/__tests__/visitors/__snapshots__/appsync-javascript-visitor.test.ts.snap

```diff
@@ -28,7 +28,7 @@ export declare class Comment {
   };
   readonly customPostId: string;
   readonly content: string;
-  readonly post?: Post | null;
+  readonly post?: (LazyLoading extends Disabled ? Post : Promise<Post>) | null;
```

Comment:
The return types need to be optional only if Lazy loading is disabled. So, I think we need to change this to something like:

```
  readonly post: LazyLoading extends Disabled ? (Post | null | undefined) : Promise<Post | undefined>;
```

Basically, the field should not appear optional if lazy loading is enabled, but the return type from the promise could possibly be missing.

@manueliglesias, correct me if I'm wrong, but I think we also decided these promises should resolve to `undefined` instead of `null` for missing values. Is that right?

(I've left `null` in the eager version for consistency with the old types.)